### DISCUSSION TITLE
RN: onTouched and isValid/isDirty

### DISCUSSION THREAD

**Discussion by jorjimen** [react-hook-form/react-hook-form#9040] [Q&A]
Currently working on RN:

I have two `TextInput`, each wrapped in their own `Controller `with their `onBlur` and `onChange `correctly assigned. Also on the `formState` declaration, I'm using `errors`, `isValid` and `isDirty`.

According to the documentation for `mode: 'onTouched'`: "Validation will trigger on the first blur event. After that, it will trigger on every change event."

At the bottom I have a `Button` with `disabled={{!isValid || !isDirty}}`.

The situation is that if I click into the first `TextInput` and type in a correct parameter, then I click into the second input, and also type in a correct parameter, the `Button` is not enabled because the form is not being re-validated until I either click enter on my keyboard or until I go back into the other TextInput.

I'm wondering if I'm understanding the behavior incorrectly, but shouldn't it be the case that if I click on the first `TextInput`, type in a valid parameter, and then I click on the second input, this input should be validated when I change every character? It doesn't seem to be validating this way.

If I am misunderstanding how onTouched works, please let me know! Basically, the behavior I want to achieve is that the first input is only validated onBlur, and then after this, every input is validated onChange.

**Comment by bluebill1049** [comment]
could you make sure onBlur is assigned to `useController/Controller`? This is key to notify hook from field is been touched.  also maybe worth to reproduce the issue in a codesandbox or snack.
